### ISSUE TITLE
Feature Request: Configurable field type/schema 

### ISSUE THREAD

**Issue by fthues** [codemonauts/craft-readonly-field#1]
Thanks for the plugin!

It would be great to be able to define the database column type of the readonly field. Right now it's hardcoded to `Schema::TYPE_STRING`, but we need `Schema::TYPE_INTEGER` instead.

**Comment by kringkaste** [comment]
Yeah, we've been thinking about that, too. But then you will have to offer Float as well. And in both cases you need a way to define the format for the output, right?

**Comment by fleaz** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [readonly-field: Andere Datentypen](https://trello.com/c/qxHUuD6G/27-readonly-field-andere-datentypen)

**Comment by skeemer** [comment]
This issue is pretty old, but it helped me figure out how to create the field I needed (Schema::TYPE_TEXT) for content length. I just extended the class with what I needed. I'm throwing this here in case someone else needs something similar.

```php
<?php

namespace modules\fields;

use Craft;
use yii\db\Schema;

class ReadonlyTextField extends \codemonauts\readonly\fields\ReadonlyField
{
    public string $columnType = Schema::TYPE_TEXT;

    public static function displayName(): string
    {
        return Craft::t('app', 'Read-only Field (Large)');
    }
}
```
